pkg/utils/functional/functional.go

```diff
@@ -112,12 +112,12 @@ func InvertStringMap(stringMap map[string]string) map[string]string {
 }
 
 // MaxDuration returns the largest duration
-func MaxDuration(durations ...time.Duration) time.Duration {
-	var max time.Duration
+func MinDuration(durations ...time.Duration) time.Duration {
+	var min time.Duration
 	for _, duration := range durations {
-		if duration > max {
-			max = duration
+		if (duration < min && duration != 0) || (duration != 0 && min == 0) {
```

Comment:
Just for safety, clarity, can you use https://pkg.go.dev/time#Time.IsZero ?

Reply:
Yes! Thank you.

Reply:
Since we're comparing `time.Duration{}` and not `time.Time{}`, this kind of comparison should be okay. 

From their code 
`type Duration int64`

Reply:
Yeah, I realize it works either way, but I think IsZero is preferable from a readability standpoint.

Reply:
So since there's not a `time.Duration.IsZero()` function, are you suggesting I add it?